lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,275 @@
+    #   Response if sections are not supported:
+    #     {"code": 501, "error": "Sections are not supported"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_group(group_name)
+      raise "group_name must be provided" if group_name.blank?
+      parse get("/groups/#{URI.escape(group_name)}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain group '#{group_name}' from External IPAM."))
+    end
+
+    # Get a list of subnets for the given External IPAM section.
+    #
+    # Params:  1. section_name: The name of the External IPAM section
+    #
+    # Returns: An array of subnets on success, or a hash with a "message" key
+    #          containing error on failure.
+
+    # Examples:
+    #   Response if success: {"code": 200, "success": true, "data": [
+    #     {"id":"9","subnet":"100.10.10.0","mask":"24","sectionId":"5","description":"Test Subnet 1"},
+    #     {"id":"10","subnet":"100.20.20.0","mask":"24","sectionId":"5","description":"Test Subnet 2"}
+    #   ],"time":0.018}
+    #   Response if no subnets exist in section.
+    #     {"code": 200, "success": true, "data": [], "time": 0.007}
+    #   Response if section not found:
+    #     {"code": 404, "error": "Section not found in External IPAM."}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_subnets_by_group(group_name)
+      raise "group_name must be provided" if group_name.blank?
+      group = URI.unescape(group_name)
+      parse get("/groups/#{URI.escape(group)}/subnets")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnets in group %s from External IPAM."), group_name)
+    end
+
+    # Returns an array of subnets from External IPAM. If the "Require unique subnets" setting in External IPAM
+    # is enabled, this will return one subnet(or zero if not exists). If this setting is disabled(i.e. you can
+    # have duplicate subnets and IP's), then it is possible to get more than one.
+    #
+    # Params:  1. subnet: The IPv4 or IPv6 subnet CIDR. (Examples: IPv4 - "100.10.10.0/24", IPv6 - "2001:db8:abcd:12::/124")
+    #
+    # Returns: A subnet with "data" key, or a hash with a "message" key
+    #          containing error on failure.
+    #
+    # Examples:
+    #   Response if subnet(s) exists:
+    #     {"code": 200, "success": true, "data": [
+    #       {"id": "32", "subnet": "10.10.10.0", "description": "Test", "mask": "29"}
+    #     ], "time": 0.007}
+    #   Response if subnet not exists:
+    #     {"code": 200, "error": "No subnets found"}
+    #   Response if can't connect to External IPAM server
+    #     {"code": 500, "error": "Unable to connect to External IPAM server"}
+    def get_subnet(subnet, externalipam_group)
```

Comment:
Docs missing the new group param. Also, I don't see "Require unique subnets" anywhere in the codebase, is this something to be implemented? Or is this a setting in the plugin?

Reply:
`Require unique subnets` is a setting in phpIPAM itself. If enabled, there can be no duplication or overlap of IP's and/or subnets. If disabled, duplication is allowed across sections in phpIPAM